components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1098,31 +1108,68 @@ public static String getFilename(int series, int image, IFormatReader r,
      MetadataStore store = r.getMetadataStore();
      MetadataRetrieve retrieve = store instanceof MetadataRetrieve ?
        (MetadataRetrieve) store : new DummyMetadata();
-     return getFilename(series, image, retrieve, pattern, padded);
+     MetadataOptions options = r.getMetadataOptions();
+     int baseIndex = 0;
+     if (options instanceof DynamicMetadataOptions) {
+       baseIndex = ((DynamicMetadataOptions) options).getInteger(BASE_INDEX, 0);
+     }
+     return getFilename(series, image, retrieve, pattern, padded, baseIndex);
   }
 
   public static String getFilename(int series, int image, MetadataRetrieve retrieve,
       String pattern, boolean padded) throws FormatException, IOException
   {
+    return getFilename(series, image, retrieve, pattern, padded, 0);
+  }
+
+  public static String getFilename(int series, int image, MetadataRetrieve retrieve,
+      String pattern, boolean padded, int indexBase) throws FormatException, IOException
+  {
+    String filename = pattern;
+    int sizeC = retrieve.getChannelCount(series);
+    int sizeT = retrieve.getPixelsSizeT(series).getValue();
+    int sizeZ = retrieve.getPixelsSizeZ(series).getValue();
+    int seriesCount = retrieve.getImageCount();
+    String imageName = retrieve.getImageName(series);
+    
+    DimensionOrder order = retrieve.getPixelsDimensionOrder(series);
+    int[] coordinates = FormatTools.getZCTCoords(order.getValue(), sizeZ, sizeC, sizeT, sizeZ*sizeC*sizeT, image);
+    String channelName = retrieve.getChannelName(series, coordinates[1]);
+    if (channelName == null) channelName = String.valueOf(coordinates[1]);
+    
+    String[] conditionalStrings = substringsBetween(filename, "[", "]");
+    for (int i = 0; i < conditionalStrings.length; i++) {
+      String subString = conditionalStrings[i];
+      boolean removeSubstring = false;
+      if (subString.contains(SERIES_NUM) && seriesCount == 1) removeSubstring = true;
+      if (subString.contains(SERIES_NAME) && (imageName == null || imageName.isEmpty())) removeSubstring = true;
+      if (subString.contains(CHANNEL_NUM) && sizeC == 1) removeSubstring = true;
+      if (subString.contains(CHANNEL_NAME) && (channelName == null || channelName.isEmpty())) removeSubstring = true;
+      if (subString.contains(CHANNEL_SIZE) && sizeC == 1) removeSubstring = true;
+      if (subString.contains(T_NUM) && sizeT == 1) removeSubstring = true;
+      if (subString.contains(T_SIZE) && sizeT == 1) removeSubstring = true;
+      if (subString.contains(Z_NUM) && sizeZ == 1) removeSubstring = true;
+      if (subString.contains(Z_SIZE) && sizeZ == 1) removeSubstring = true;
+      if (removeSubstring) {
+        filename = filename.replace("["+subString+"]", "");
+      }
+      else {
+        filename = filename.replace("["+subString+"]", subString);
```

Comment:
It looks like this changes the behavior when square brackets are included in the output filename - is that by design?

With develop alone:

```
$ bfconvert curated/biorad-gel/samples/imagesEight.1sc "images[8].tiff"
$ ls *.tiff
images[8].tiff
```
with this PR:

```
$ bfconvert curated/biorad-gel/samples/imagesEight.1sc "images[8].tiff"
$ ls *.tiff
images8.tiff
```

Could the new syntax be documented, so that it's clear what the intended/expected behavior is?